onCmdDirUsage is linux-only for now

No tar -a, use archivers/gtar

Index: src/FilePanel.cpp
--- src/FilePanel.cpp.orig
+++ src/FilePanel.cpp
@@ -128,7 +128,6 @@ FXDEFMAP(FilePanel) FilePanelMap[] =
     FXMAPFUNC(SEL_COMMAND, FilePanel::ID_RUN_SCRIPT, FilePanel::onCmdRunScript),
     FXMAPFUNC(SEL_UPDATE, FilePanel::ID_RUN_SCRIPT, FilePanel::onUpdRunScript),
     FXMAPFUNC(SEL_COMMAND, FilePanel::ID_GO_SCRIPTDIR, FilePanel::onCmdGoScriptDir),
-    FXMAPFUNC(SEL_COMMAND, FilePanel::ID_DIR_USAGE, FilePanel::onCmdDirUsage),
     FXMAPFUNC(SEL_RIGHTBUTTONRELEASE, FilePanel::ID_FILELIST, FilePanel::onCmdPopupMenu),
     FXMAPFUNC(SEL_COMMAND, FilePanel::ID_POPUP_MENU, FilePanel::onCmdPopupMenu),
     FXMAPFUNC(SEL_DOUBLECLICKED, FilePanel::ID_FILELIST, FilePanel::onCmdItemDoubleClicked),
@@ -159,6 +158,7 @@ FXDEFMAP(FilePanel) FilePanelMap[] =
     FXMAPFUNC(SEL_UPDATE, FilePanel::ID_TOGGLE_THUMBNAILS, FilePanel::onUpdToggleThumbnails),
     FXMAPFUNC(SEL_UPDATE, FilePanel::ID_DIR_USAGE, FilePanel::onUpdDirUsage),
 #if defined(linux)
+    FXMAPFUNC(SEL_COMMAND, FilePanel::ID_DIR_USAGE, FilePanel::onCmdDirUsage),
     FXMAPFUNC(SEL_COMMAND, FilePanel::ID_MOUNT, FilePanel::onCmdMount),
     FXMAPFUNC(SEL_COMMAND, FilePanel::ID_UMOUNT, FilePanel::onCmdMount),
     FXMAPFUNC(SEL_UPDATE, FilePanel::ID_MOUNT, FilePanel::onUpdMount),
@@ -4984,6 +4984,12 @@ long FilePanel::onCmdAddToArch(FXObject* o, FXSelector
             cmd = "tar -acvf " + archive + " ";
         }
 
+        // Change tar to gtar
+        if (compare(cmd, "tar ", 4) == 0)
+        {
+            cmd.prepend('g');
+        }
+
         for (int u = 0; u < current->list->getNumItems(); u++)
         {
             if (current->list->isItemSelected(u))
@@ -5127,6 +5133,12 @@ long FilePanel::onCmdExtract(FXObject*, FXSelector, vo
             cmd = "tar -axvf ";
         }
 
+        // Change tar to gtar
+        if (compare(cmd, "tar ", 4) == 0)
+        {
+            cmd.prepend('g');
+        }
+
         // Final extract command
         cmd += name + " ";
 
@@ -5279,6 +5291,12 @@ long FilePanel::onCmdExtractToFolder(FXObject*, FXSele
             cmd = "tar -axvf ";
         }
 
+        // Change tar to gtar
+        if (compare(cmd, "tar ", 4) == 0)
+        {
+            cmd.prepend('g');
+        }
+
         // Final extract command
         cmd += pathname + " ";
 
@@ -5385,6 +5403,12 @@ long FilePanel::onCmdExtractHere(FXObject*, FXSelector
             else
             {
                 cmd = "tar -axvf ";
+            }
+
+            // Change tar to gtar
+            if (compare(cmd, "tar ", 4) == 0)
+            {
+                cmd.prepend('g');
             }
 
             // Final extract command
